homeassistant/components/hdmi_cec.py

```diff
@@ -328,30 +328,34 @@ def __init__(self, hass: HomeAssistant, device, logical) -> None:
         self._state = STATE_UNKNOWN
         self._logical_address = logical
         self.entity_id = "%s.%d" % (DOMAIN, self._logical_address)
-        device.set_update_callback(self._update)
 
     def update(self):
         """Update device status."""
-        self._update()
+        device = self._device
+        from pycec.const import STATUS_PLAY, STATUS_STOP, STATUS_STILL, \
+            POWER_OFF, POWER_ON
+        if device.power_status == POWER_OFF:
+            self._state = STATE_OFF
+        elif device.status == STATUS_PLAY:
+            self._state = STATE_PLAYING
+        elif device.status == STATUS_STOP:
+            self._state = STATE_IDLE
+        elif device.status == STATUS_STILL:
+            self._state = STATE_PAUSED
+        elif device.power_status == POWER_ON:
+            self._state = STATE_ON
+        else:
+            _LOGGER.warning("Unknown state: %d", device.power_status)
+
+    async def async_added_to_hass(self):
+        """Register callbacks after initialization."""
+        self._device.set_update_callback(self._update)
+        self.update()
```

Comment:
We can remove this. It will be called during entity addition.

Reply:
@MartinHjelmare I'm using this to register device callback. Should I remove only `self.update()`?

Reply:
Yes, just that line. The callback registration looks perfect.